Get task list

jahto opened the pull request:
This is what I got finished today. Continuing tomorrow, sometimes it's time to just say this is enough for this day, stop working and have some free time (and some beers...).

There are still some things I'll have to check about anonymous users and how a preference 'anon_group' is supposed to work when combined with a project preference others_view, which is actually anyone can see.

Sill to-do: group preferences view_own_tasks and view_groups_tasks. The first one is easy, almost the same check as done for private tasks. The second one needs some more thinking, have to get somehow to the other users in group.

By the way, we have some serious problems with global and project specific permissions. Tested situation where I'm in global group developers, have rights to view everything and manage every project. Then as admin I added 1 special project that I would not like to even see when I'm logged in with my normal account, so I added a new group "Forbidden" that has not any rights whatever to that special and added my normal account to it. Fine with task list, tasks belonging to that project don't show up. But it still showed up in project list and I could even manage the project. Not good. In my opinion, anything that is specifically overridden on project level should have the final say, unless the user really is global admin, then no restrictions at all. What's your opinion?

peterdd committed:
enhancing readablility of sql for get_task_list()

This is our workhorse, we should be able to see quick what is happening.

jahto committed:
Even smaller result set, just one row.

jahto committed:
Merge branch 'patch-1' of git://github.com/peterdd/flyspray into get_task_list

jahto committed:
Add view permission check for anonymous user.

jahto committed:
By just using a few joins we can get rid of a lot of data user is not allowed to see, but not all yet.

jahto committed:
Better commenting.

jahto committed:
Handle private tasks.

jahto committed:
Missing condition when joining to groups table.

Commit:
Merge pull request #16 from Flyspray/master

Merge master

jahto committed:
Radical change in how permission are checked. Project level permission can now override global ones.

jahto committed:
Removed now unnecessary code. Have to check how anonymous users should be handled.

peterdd commented:
There was in the old wiki big article about how the permissions work.

If I remember, it works like global permissions, then ADD permissions by project. You cannot DENY permissions on project level.

If I understood right:  If you want that very strict permission restriction on a project even for global developers, you must drop that developers from the global developer group and maybe other users from other global groups or drop global permissions on global groups.

Then add project groups and project group permissions and add people to that project groups.

Sorry I still didn't implemented the permission view , which I have on paper but everytime we stumble over much more things to fix too.

I personal think this is not a bad permission layout. Probably enables easier/faster SQL queries respecting permissions than other solutions. (look user perms array)

peterdd commented:
By implementing the permissions view, I had the idea of geen and yellow check icons or a slight 3D angle view with 2 layers (global and project permission layer)
Both checks means access granted, but one of the meant grant by global permission, even if on project level the permission is not checked.

jahto committed:
Broke admin login... not final, only temporary solution.

jahto committed:
Final change, works now according to our manual, not to my taste or needs.

peterdd committed:
Update theme.css

peterdd committed:
Update theme.css

Commit:
Merge pull request #219 from peterdd/patch-2

buttons on quickaction menu

Commit:
Merge pull request #17 from Flyspray/master

Merge master

Commit:
Composer installation fix

Ok, this is working again... Also cleaned up the code, went for code
conventions etc etc etc

Commit:
Merge pull request #220 from Psychokiller1888/master

Fix composer install

Commit:
These buttons are useless

According to W3C onchange events are supported by EVERY browser.
Security disabling them is none of FS business

jahto committed:
Add parenthesis for mysql.

Commit:
Fix button in details_view

As well as the spacing in the task details and code identation...

Commit:
Merge pull request #221 from Psychokiller1888/master

Fix details buttons

jahto committed:
Use correct datatype and actual value, and database index might even be used by db engine.

Commit:
Forgot make public

And that entire quick edit is a CHAOS with those HUGE buttons for blind
people

Commit:
Merge pull request #222 from Psychokiller1888/master

Forgot make public

jahto committed:
2 cases out of 3 now handled on database side.

jahto committed:
Non-private tasks in projects everyone is allowed to see now handled.

jahto committed:
Merge remote-tracking branch 'upstream/master' into get_task_list

jahto committed:
Forgot 2 fields when inserting/updating a group.

Commit:
Merge pull request #223 from jahto/misc_small_fixes

Forgot 2 fields when inserting/updating a group.

Commit:
Merge pull request #18 from Flyspray/master

Merge master

jahto committed:
Handled view_groups_tasks permission in dababase. That was a mind-boggling monster.

jahto committed:
Moved a few lines and removed now unnecessary code.

jahto committed:
Changed comment.

Commit:
Fix huge mess in quick edit mode

Took me about 15 minutes to make something aligned and looking ok...
Check twitch for replay

Commit:
Merge pull request #224 from Psychokiller1888/master

Fixing button mess in quick edit mode

jahto committed:
Merge remote-tracking branch 'upstream/master' into get_task_list

peterdd commented:
I will use your branch for testing your changes.

Maybe we first merge the memory reducing patch without the limit offset in sql. This breaks IMHO nothing in the existing master.

Your work on putting the whole permission system into the sql query is honourable, but the changes are huge and needs much testing and with many corner cases. 

Damn, I can't test that all as fast as you and psycho add changes...

peterdd commented:
I will use your branch for testing your changes.

Maybe we first merge the memory reducing patch without the limit offset in sql. This breaks IMHO nothing in the existing master.

Your work on putting the whole permission system into the sql query is honourable, but the changes are huge and needs much testing and with many corner cases.

Damn, I can't test that all as fast as you and psycho add changes...
